modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/CreateAPIApplication.testcase

```diff
@@ -0,0 +1,324 @@
+@component-name = "portal-headless"
+definition {
+
+	property custom.properties = "feature.flag.LPS-184413=true${line.separator}feature.flag.LPS-167253=true${line.separator}feature.flag.LPS-178642=true${line.separator}feature.flag.LPS-186757=true";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Object";
+	property testray.main.component.name = "Headless Builder";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginUI();
+
+		task ("Given navigate to Object > API Builder") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Object",
+				panel = "Control Panel",
+				portlet = "API Builder");
+		}
+
+		task ("And Given click New button to add one API") {
+			Click(locator1 = "Button#PLUS");
+		}
+	}
+
+	tearDown {
+		ApplicationAPI.deleteAllAPIApplication();
+
+		var testLiferayVirtualInstance = PropsUtil.get("test.liferay.virtual.instance");
+
+		if (${testLiferayVirtualInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@priority = 4
+	test CanAutoGenerateUrlBasedOnTitle {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+
+		task ("When I enter title "M*y Title 123"") {
+			Type(
+				key_fieldLabel = "Title",
+				locator1 = "Modal#INPUT_WITH_LABEL",
+				value1 = "M*y Title 123");
+		}
+
+		task ("Then the automated url is my-title-123") {
+			AssertTextEquals(
+				locator1 = "APIBuilder#AUTOMATED_URL",
+				value1 = "my-title-123");
+		}
+	}
+
+	@priority = 4
+	test CanCreateAPIOnVirtualInstance {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+
+		task ("And Given create an API with title "test"") {
+			ApplicationAPI.createAPIApplication(
+				baseURL = "test",
+				status = "unpublished",
+				title = "test");
+		}
+
+		task ("And Given create a new virtual instance") {
+			HeadlessPortalInstanceAPI.addPortalInstance(
+				domain = "liferay.com",
+				portalInstanceId = "www.able.com",
+				virtualHost = "www.able.com");
+		}
+
+		task ("When I create an API with title "test" on the virtual instance") {
+			User.firstLoginUI(
+				password = "test",
+				specificURL = "http://www.able.com:8080",
+				userEmailAddress = "test@liferay.com");
+
+			APIBuilderUI.createAPIApplication(title = "test");
+		}
+
+		task ("Then the test API is created successfully") {
+			Alert.viewSuccessMessageText(successMessage = "New API Application was created.");
+
+			AssertElementPresent(
+				key_name = "test",
+				locator1 = "ObjectAdmin#TABLE_LIST_TITLE");
+		}
+	}
+
+	@priority = 4
+	test CanCreateApiWithValidtFieldValues {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+
+		task ("When I create an API with title "M*y Title 123" and some description") {
```

Comment:
Could we use APIBuilderUI.createAPIApplication() here?

Reply:
ok, a good suggestion, will update